### PR TITLE
Support for packed_array

### PR DESCRIPTION
**Pull request by alaindargelas**

### CODE REVIEW

**Comment by alaindargelas** [comment]
@kamilrakoczy , this is ready for review.

File: frontends/systemverilog/uhdm_ast.cc [hunk 1]
```diff
@@ -4780,6 +4834,7 @@ void UhdmAst::process_port()
             delete node;
         }
     });
+
```
Comment (kgugala) [inline frontends/systemverilog/uhdm_ast.cc:4837]: ditto

File: frontends/systemverilog/uhdm_ast.cc [hunk 2]
```diff
@@ -1822,14 +1822,21 @@ void UhdmAst::process_packed_array_typespec()
     current_node = make_ast_node(AST::AST_WIRE);
     visit_one_to_many({vpiRange}, obj_h, [&](AST::AstNode *node) { packed_ranges.push_back(node); });
     visit_one_to_one({vpiElemTypespec}, obj_h, [&](AST::AstNode *node) {
-        if (node && node->type == AST::AST_STRUCT) {
+        if (node && (node->type == AST::AST_STRUCT)) {
```
Comment (kgugala) [inline frontends/systemverilog/uhdm_ast.cc:1825]: this looks like formatting change, please do not mix formatting and functional changes in single commit

**Review by kgugala — CHANGES_REQUESTED** [review]

File: frontends/systemverilog/uhdm_ast.cc [hunk 3]
```diff
@@ -2839,6 +2846,43 @@ void UhdmAst::process_array_var()
...
+                            // anonymous typespec, move the children to variable
+                            current_node->type = node->type;
+                            current_node->children = std::move(node->children);
+                        } else {
+                            auto wiretype_node = new AST::AstNode(AST::AST_WIRETYPE);
+                            wiretype_node->str = node->str;
+                            current_node->children.push_back(wiretype_node);
+                            current_node->is_custom_type = true;
+                        }
+                        delete node;
```
[the remark this answers is not in the document]
Comment (alaindargelas) [reply frontends/systemverilog/uhdm_ast.cc:2879]: This is the way these Lamda are all created. Check existing code line: 2823-2826 and 2838-2841. There are many pre-existing mini code snippet dupplication all over in the lambas. What is special about this one?

There is no nice way to describe what these 4 lines do in a function.

File: frontends/systemverilog/uhdm_ast.cc [hunk 4]
```diff
@@ -1822,14 +1822,21 @@ void UhdmAst::process_packed_array_typespec()
...
             node->cloneInto(current_node);
             current_node->str = str;
             delete node;
+        } else if (node && (node->type == AST::AST_ENUM)) {
+            AST::AstNode *const wiretype_node = make_named_node(AST::AST_WIRETYPE);
+            wiretype_node->str = node->str;
+            current_node->children.push_back(wiretype_node);
+            current_node->is_custom_type = true;
+            current_node->str = node->str;
+            delete node;
```
[the remark this answers is not in the document]
Comment (alaindargelas) [reply frontends/systemverilog/uhdm_ast.cc:1839]: It is not. The lamba payloads look similar but they are not exactly the same. Usage of Lamba is pervasive throughout the code, with very litle function factorization. This code is similar to existing code.

File: frontends/systemverilog/uhdm_ast.cc (hunk 3 above)
Comment (kgugala) [inline frontends/systemverilog/uhdm_ast.cc:2879]:
```suggestion
                 auto vpi_obj = vpi_get(vpiType, reg_h2);
                 if (vpi_obj == vpiStructVar || vpi_obj == vpiEnumVar || vpi_obj == vpiLogicVar) {
                     if (vpi_obj == vpiLogicVar)
                         current_node->is_logic = true;
                     visit_one_to_one({vpiTypespec}, reg_h2, [&](AST::AstNode *node) {
                         if (node->str.empty()) {
                             // anonymous typespec, move the children to variable
                             current_node->type = node->type;
                             current_node->children = std::move(node->children);
                         } else {
                             auto wiretype_node = new AST::AstNode(AST::AST_WIRETYPE);
                             wiretype_node->str = node->str;
                             current_node->children.push_back(wiretype_node);
                             current_node->is_custom_type = true;
                         }
                         delete node;
```

With this we can avoid code duplication + `vpi_get` will be called only once (unless I'm missing something)

**Review by kgugala — CHANGES_REQUESTED** [review]

**Review by kgugala — APPROVED** [review]

**Pull request merged**